extensions/amp-story/1.0/amp-story-interactive-confetti.js

```diff
@@ -0,0 +1,124 @@
+ * limitations under the License.
+ */
+
+import {htmlFor} from '../../../src/static-template';
+import {removeChildren} from '../../../src/dom';
+import {setStyles} from '../../../src/style';
+
+/**
+ * Generates the template for the particle wrapper.
+ *
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildWrapperTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <div
+      class="i-amphtml-story-interactive-particle-wrapper"
+      aria-hidden="true"
+    ></div>
+  `;
+};
+
+/**
+ * Generates a template for a particle.
+ *
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildParticleTemplate = (element) => {
+  const html = htmlFor(element);
+  return html` <div class="i-amphtml-story-interactive-particle"></div> `;
+};
+
+/**
+ * Creates a radial burst of emoji.
+ * The burst begins from the center of the parent element and
+ * animates to the edges of the parent element.
+ * Nodes are removed from the dom at the end of the animation.
+ *
+ * @param {!Element} rootEl
+ * @param {string} confettiEmoji
+ * @return {void}
+ */
+export function emojiBurst(rootEl, confettiEmoji) {
+  const PARTICLE_COUNT = 5;
+  const SLICE = (Math.PI * 2) / PARTICLE_COUNT;
+
+  const ANGLE_RANDOMNESS = SLICE * 0.2;
+  const FONT_SIZE_RANDOM_RANGE = [30, 50];
+  const ROTATION_RANDOMNESS = 20;
+  const ADDITIONAL_DISTANCE = 5;
+
+  const ANIMATION_IN_DELAY = 300;
+  const ANIMATION_OUT_DELAY = 1000;
+
+  // To calculate particle transform distance.
+  const ROOT_EL_RECT = rootEl./*OK*/ getBoundingClientRect();
+
+  const particleWrapper = buildWrapperTemplate(rootEl);
+  rootEl.appendChild(particleWrapper);
+
+  setTimeout(() => {
+    // Generate particles. Set their ending position, size and rotation.
+    for (let i = 0; i < PARTICLE_COUNT; i++) {
+      const particle = buildParticleTemplate(rootEl);
+      particleWrapper.appendChild(particle);
+      particle.textContent = confettiEmoji;
+
+      const fontSize =
+        randomInRange(FONT_SIZE_RANDOM_RANGE[0], FONT_SIZE_RANDOM_RANGE[1]) +
+        'px';
+      const angle =
+        SLICE * i + randomInRange(-ANGLE_RANDOMNESS, ANGLE_RANDOMNESS);
+      const x =
+        Math.sin(angle) * (ROOT_EL_RECT.width / 2 + ADDITIONAL_DISTANCE);
+      const y =
+        Math.cos(angle) * (ROOT_EL_RECT.height / 2 + ADDITIONAL_DISTANCE);
+      const rotation = randomInRange(-ROTATION_RANDOMNESS, ROTATION_RANDOMNESS);
+      setStyles(particle, {
+        fontSize,
+        transform: `translate(${x}px, ${y}px) rotate(${rotation}deg)`,
+      });
+    }
+    // Scale up particle container.
+    setStyles(particleWrapper, {transform: 'scale(1)'});
+
+    // Animate out the particle wrapper and remove particles from the dom.
+    setTimeout(() => {
+      setStyles(particleWrapper, {
+        transform: `scale(1.2)`,
+        opacity: `0`,
+      });
+
+      particleWrapper.addEventListener('transitionend', () => {
```

Comment:
Coverage looks [good](https://caniuse.com/#feat=mdn-api_htmlelement_transitionend_event) but I'm changing this to a timer since that's the pattern used elsewhere in the method.